examples/postgres/main.go

```diff
@@ -0,0 +1,390 @@
+
+	return &PostgresServer{
+		db:              db,
+		resourceBaseURL: resourceBaseURL,
+	}, nil
+}
+
+// Close closes the database connection
+func (ps *PostgresServer) Close() error {
+	return ps.db.Close()
+}
+
+// ListTables returns all tables in the public schema as resources
+func (ps *PostgresServer) ListTables(ctx context.Context) ([]*mcp.ServerResource, error) {
+	query := "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'"
+	rows, err := ps.db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query tables: %w", err)
+	}
+	defer rows.Close()
+
+	var resources []*mcp.ServerResource
+	for rows.Next() {
+		var tableName string
+		if err := rows.Scan(&tableName); err != nil {
+			return nil, fmt.Errorf("failed to scan table name: %w", err)
+		}
+
+		resourceURI := fmt.Sprintf("%s/%s/%s", ps.resourceBaseURL.String(), tableName, SCHEMA_PATH)
+		resource := &mcp.ServerResource{
+			Resource: &mcp.Resource{
+				URI:      resourceURI,
+				MIMEType: "application/json",
+				Name:     fmt.Sprintf(`"%s" database schema`, tableName),
+			},
+			Handler: ps.readTableSchema,
+		}
+		resources = append(resources, resource)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("error iterating tables: %w", err)
+	}
+
+	return resources, nil
+}
+
+// readTableSchema handles reading table schema information
+func (ps *PostgresServer) readTableSchema(ctx context.Context, ss *mcp.ServerSession, params *mcp.ReadResourceParams) (*mcp.ReadResourceResult, error) {
+	resourceURL, err := url.Parse(params.URI)
+	if err != nil {
+		return nil, fmt.Errorf("invalid resource URI: %w", err)
+	}
+
+	pathComponents := []string{}
+	for _, component := range []string{resourceURL.Path} {
+		if component != "" {
+			pathComponents = append(pathComponents, component)
+		}
+	}
+
+	// Parse path: /tableName/schema
+	parts := []string{}
+	for _, part := range pathComponents {
+		if part == "/" {
+			continue
+		}
+		subParts := []string{}
+		for _, subPart := range []string{part} {
+			if subPart != "" {
+				for _, p := range []string{subPart} {
+					if p != "/" {
+						subParts = append(subParts, p)
+					}
+				}
+			}
+		}
+		parts = append(parts, subParts...)
+	}
+
+	// Extract table name and schema path from URI
+	urlPath := resourceURL.Path
+	if urlPath == "" {
+		return nil, fmt.Errorf("empty URI path")
+	}
+
+	// Remove leading slash and split by '/'
+	trimmedPath := urlPath
+	if trimmedPath[0] == '/' {
+		trimmedPath = trimmedPath[1:]
+	}
+	pathParts := []string{}
+	for _, part := range []string{trimmedPath} {
+		if part != "" {
+			// Split by '/'
+			for i, p := range []string{part} {
+				if i == 0 {
+					subparts := []string{}
+					current := ""
+					for _, r := range p {
+						if r == '/' {
+							if current != "" {
+								subparts = append(subparts, current)
+								current = ""
+							}
+						} else {
+							current += string(r)
```

Comment:
This can be quadratic. Consider using a []string and calling strings.Join at the end.